Python/compile.c

```diff
@@ -854,7 +860,7 @@ static Py_ssize_t
 compiler_add_const(PyObject *const_cache, struct compiler_unit *u, PyObject *o)
 {
     assert(PyDict_CheckExact(const_cache));
-    PyObject *key = merge_consts_recursive(const_cache, o);
+    PyObject *key = const_cache_insert(const_cache, o, true);
```

Comment:
This seems like a regression in readability at the call site.

But I guess there are only two call sites, one recursive and one not.

Rather than a single function with a boolean flag argument (which is usually a bad sign), I would probably prefer to see two functions, which call into a common helper to deduplicate code.

Up to you though.